vpr/src/route/rr_graph_indexed_data.cpp

```diff
@@ -264,119 +260,104 @@ static std::vector<size_t> count_rr_segment_types() {
-
-    int inode, itrack, cost_index;
-    float Tdel, Tdel_sum, frac_num_seg;
-
     auto& device_ctx = g_vpr_ctx.device();
+    auto& rr_indexed_data = device_ctx.rr_indexed_data;
 
-    Tdel_sum = 0.;
+    float Tdel_sum = 0.0;
+    int Tdel_num = 0;
+    for (size_t cost_index = CHANX_COST_INDEX_START; cost_index < rr_indexed_data.size(); cost_index++) {
+        float inv_length = device_ctx.rr_indexed_data[cost_index].inv_length;
+        float T_value = rr_indexed_data[cost_index].T_linear * inv_length + rr_indexed_data[cost_index].T_quadratic * std::pow(inv_length, 2);
 
-    for (itrack = 0; itrack < nodes_per_chan; itrack++) {
-        inode = find_average_rr_node_index(device_ctx.grid.width(), device_ctx.grid.height(), CHANX, itrack,
-                                           L_rr_node_indices);
-        if (inode == -1)
-            continue;
-        cost_index = device_ctx.rr_nodes[inode].cost_index();
-        frac_num_seg = clb_dist * device_ctx.rr_indexed_data[cost_index].inv_length;
-        Tdel = frac_num_seg * device_ctx.rr_indexed_data[cost_index].T_linear
-               + frac_num_seg * frac_num_seg
-                     * device_ctx.rr_indexed_data[cost_index].T_quadratic;
-        Tdel_sum += Tdel / (float)clb_dist;
+        if (T_value == 0.0) continue;
+
+        Tdel_sum += T_value;
+        Tdel_num += 1;
     }
 
-    for (itrack = 0; itrack < nodes_per_chan; itrack++) {
-        inode = find_average_rr_node_index(device_ctx.grid.width(), device_ctx.grid.height(), CHANY, itrack,
-                                           L_rr_node_indices);
-        if (inode == -1)
-            continue;
-        cost_index = device_ctx.rr_nodes[inode].cost_index();
-        frac_num_seg = clb_dist * device_ctx.rr_indexed_data[cost_index].inv_length;
-        Tdel = frac_num_seg * device_ctx.rr_indexed_data[cost_index].T_linear
-               + frac_num_seg * frac_num_seg
-                     * device_ctx.rr_indexed_data[cost_index].T_quadratic;
-        Tdel_sum += Tdel / (float)clb_dist;
+    float delay_norm_fac = Tdel_sum / Tdel_num;
+
+    if (getEchoEnabled() && isEchoFileEnabled(E_ECHO_RR_GRAPH_INDEXED_DATA)) {
+        std::ofstream out_file;
+
+        out_file.open(getEchoFileName(E_ECHO_RR_GRAPH_INDEXED_DATA));
+        out_file << "Delay normalization factor: " << delay_norm_fac << std::endl;
+
+        out_file.close();
     }
 
-    return (Tdel_sum / (2. * nodes_per_chan));
+    return delay_norm_fac;
 }
 
-static void load_rr_indexed_data_T_values(int index_start,
-                                          int num_indices_to_load,
-                                          t_rr_type rr_type,
-                                          int nodes_per_chan,
-                                          const t_rr_node_indices& L_rr_node_indices) {
+static void load_rr_indexed_data_T_values() {
     /* Loads the average propagation times through segments of each index type   *
```

Comment:
This comment looks out of date.  I think now you're going through all rr_nodes and getting medians for their delay. Good to update this comment and check for other out-of-date comments. Should also describe what we're trying to do here (get typical values of T_linear etc. for different wire types, which are used for things like constructing lookaheads and place delay matrices).